pkgs/development/libraries/protobuf/generic-v3-cmake.nix

```diff
@@ -93,6 +93,8 @@ let
           protobuf = self;
         });
       };
+
+      inherit abseil-cpp;
```

Comment:
I think this should be in `propagatedBuildInputs`.

Reply:
1. You're probably right, but can you explain it to me? I realized I don't totally understand why it should or why it should not be propagated.
2. Just to clarify, even if we add this to propagatedBuildInputs, I think it is still necessary to pass this along so that the Python derivation can fix up its setup.py file to refer to it directly.

Reply:
> 1. You're probably right, but can you explain it to me? I realized I don't totally understand why it should or why it should not be propagated.

Looking for uses of `absl` in the output store path finds lots of uses:
```
rg absl /nix/store/yz8ykkmlnf51dp307g744852q03n6bmz-protobuf-3.23.4/include/google/
protobuf/
...
```
I would generally propagate dependencies that show up in the interface of a package.
And for the record, this should have been done in my PR, but I missed it.

> 2. Just to clarify, even if we add this to propagatedBuildInputs, I think it is still necessary to pass this along so that the Python derivation can fix up its setup.py file to refer to it directly.

Ah, I didn't know propagation wouldn't solve this.

Reply:
Got it, that makes sense. Thank you for clarifying.

Reply:
Why did you resolve this without moving `abseil-cpp` to `propagatedBuildInputs`?

Reply:
Nevermind, I see that is already done in staging-next.